pkg/ctl/create/iamidentitymapping.go

```diff
@@ -133,6 +137,15 @@ func doCreateIAMIdentityMapping(cmd *cmdutils.Cmd, options iamIdentityMappingOpt
 		for _, identity := range identities {
 			arn := identity.ARN()
 
+			if iam.CompareIdentity(id, identity) {
```

Comment:
Hey @DameonSmith!

This check here, should have been behind the below if, right? :D :D 

I think we missed that. :)

Reply:
Or rather, I think this also needs the `&& options.NoDuplicateArns`.

Reply:
This check was to just ignore any duplicated identity entirely. Regardless of whether or not the duplicate arn flags was set. 
I was trying to avoid having this kind of pollution:
```
arn: arn1, namespace: default, groups: [group1, group2]
arn: arn1, namespace: default, groups: [group1, group2]
arn: arn1, namespace: default, groups: [group2, group1]
etc...
```
The context was specifically using eskctl in pipelines and IaC to create default roles and users for a cluster. The pipeline was run multiple times and would just input the same config over and over again. My assumption was that if a record was in the aws_auth config map it should just be ignored without an error as there's no functional difference. 

The NoDuplicateArns flag was meant to prevent silent shadowing of existing roles with different permissions. I wanted to ability to have better visibility on something like the "ReadOnlyUsers" role accidentally got the masters group added to it. 

Hopefully that makes sense, and there's not something about my assumptions that were mistaken. Let me know if I need to fix something.

Reply:
Oh I didn't realize that was a feature, Is there a reason to allow duplicates? 

Either way I can add that flag as a requirement if that's what you'd like. Should I open another PR or is there some way to open this one?